Microsoft.Research/Contracts/MsCorlib/System.Collections.ObjectModel.Collection.cs

```diff
@@ -0,0 +1,394 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Reflection;
+using System.Runtime;
+using System.Runtime.InteropServices;
+
+namespace System.Collections.ObjectModel
+{
+    // Summary:
+    //     Provides the base class for a generic collection.
+    //
+    // Type parameters:
+    //   T:
+    //     The type of elements in the collection.
+    public class Collection<T> : IList<T>, ICollection<T>, IEnumerable<T>, IList, ICollection, IEnumerable
```

Comment:
Yes, we should definitely come up with some rules and clean up all the code. Currently I just try to follow the pattern that I find in the current file.